src/control/cmd/daos_server/storage_nvme.go

```diff
@@ -236,6 +236,36 @@ func processNVMePrepReq(log logging.Logger, cfg *config.Server, iommuChecker har
 	return nil
 }
 
+// Clean hugepages for non-existent processes. Remove lockfiles for any devices specified in request
+// or config. Clean all lockfiles if none have been specified on cli or in config. Honor both allow
+// and block lists when processing removal of lockfiles.
+func cleanSpdkResources(log logging.Logger, req storage.BdevPrepareRequest, cmd *nvmeCmd) {
+	cleanReq := storage.BdevPrepareRequest{
+		CleanSpdkHugepages: true,
+		CleanSpdkLockfiles: true,
+	}
+
+	// Differentiate between an empty allow list due to filter combinations (all allowed devices
+	// also present in block list) from situation where no devices are selected in cli or config
+	// lists. In the latter case, all lockfiles should be removed.
```

Comment:
Just curious, why are the filters allowed to cancel each other out? Shouldn't conflicting allow/block lists be an invalid input?